elastic/kibana
# [UnifiedFieldList] Add new fields ingested in background with valid mappings

## Summary

This PR improves UnifiedFieldList adding a newly ingested field to be added to the list with already the right type in Discover. On top of that, it triggers a refresh of the selected DataView fields, so the new field will be available by consumers of the DataView, no reload is necessary, and also when we introduce caching, the cache will be refreshed. One more benefit, when switching data views, the field refreshing was removed. It's no longer necessary, making data view switching snappy, while the UnifiedFieldList makes sure, new fields are showing up, and the data view field cache is refreshed behind the scenes, if necessary.

### How to test

In Console ingest a new record

```
PUT field-refresh-test/_doc/1
{
  "message": "Hi"
}
```

Go to Discover and create a DataView for `field-refresh-test`

Open Kibana / Console in a second tab/window and ingest

```
PUT field-refresh-test/_doc/1
{
  "message": "Hi again!",
  "🐣": "🐔"
}
```
Refresh the search in the first window. Then you should be able to see the new field with the correct mapping in the field list.

<img width="481" alt="Bildschirmfoto 2023-12-01 um 09 49 03" src="https://github.com/elastic/kibana/assets/463851/69661190-2083-492e-8d73-cb0a81d60787">


Before this change the new field was available, but unmapped. It also works in Lens, when new fields caps are requested due to change of query / filters / time range 

Resolves #172342
Resolves #169360
Resolves #170874

Limitation: A type change won't be detected (this could be a follow up task) 

### Checklist
